src/marketplace/marketplace.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+"use strict";
+import * as fs from "fs";
+import * as path from "path";
+import * as vscode from "vscode";
+import { ModuleInfo } from "../common/moduleInfo";
+import { Utility } from "../common/utility";
+import { LocalServer } from "./localserver";
+
+export class Marketplace {
+    private panel: vscode.WebviewPanel;
+    private localServer: LocalServer;
+
+    constructor(private context: vscode.ExtensionContext, private modules?: string[]) {
+        this.localServer = new LocalServer(context, modules);
+    }
+
+    public async importModule(): Promise<any> {
+        this.localServer.startServer();
+        this.panel = vscode.window.createWebviewPanel(
+            "IoT Edge Marketplace",
+            "IoT Edge Marketplace",
+            vscode.ViewColumn.One,
+            {
+                enableCommandUris: true,
+                enableScripts: true,
+                retainContextWhenHidden: true,
+            },
+        );
+        let html = fs.readFileSync(this.context.asAbsolutePath(path.join("assets", "marketplace", "index.html")), "utf8");
+        html = html
+            .replace(/{{root}}/g, vscode.Uri.file(this.context.asAbsolutePath(".")).with({ scheme: "vscode-resource" }).toString())
+            .replace(/{{endpoint}}/g, this.localServer.getServerUri());
+        this.panel.webview.html = html;
+        this.panel.onDidDispose(() => {
+            this.panel = undefined;
+            this.localServer.stopServer();
+        });
+
+        return new Promise((resolve, reject) => {

```

Comment:
When will it happen, when user choose to import from market place. And import one module. But he/she doesn't close the panel, and import another again?

Reply:
If user click import, the panel will close. But if use not click import, there is still chance to show many page. I will handle it to make only one page to be shown.